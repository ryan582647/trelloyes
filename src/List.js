import React from 'react';
import Card from './Card'
import './List.css';

class List extends React.Component {
   renderCardContent() {
        return this.props.cards.map((card, index) => {
          return <Card 
          key={index}
          title={card.title} 
          content={card.content} 
           />
        })
      }
  render() {
      return (
    <section className='List'>
      <header className='List-header'>
        <h2>{this.props.header}</h2>
      </header>
      <div className='List-cards'>
      {this.renderCardContent()}
      </div>
    </section>
  );
}
};

export default List;