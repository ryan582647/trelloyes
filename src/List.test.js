import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import List from './List';

describe('List component maps and passes to Card correctly', () => {
    const allCardsList = [
        { title: 'First card', content: 'lorem ipsum' }, 
        { title: 'Second card', content: 'lorem ipsum' },
        { title: 'Third card', content: 'lorem ipsum' }
    ];

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render( <List key="1"
            header="First list"
            cards={allCardsList} />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const tree = renderer
        .create(<List key="1"
        header="First list"
        cards={allCardsList} />)
        .toJSON();
        expect(tree).toMatchSnapshot();  
    });
});