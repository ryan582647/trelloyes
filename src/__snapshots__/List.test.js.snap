// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component maps and passes to Card correctly renders the UI as expected 1`] = `
<section
  className="List"
>
  <header
    className="List-header"
  >
    <h2>
      First list
    </h2>
  </header>
  <div
    className="List-cards"
  >
    <div
      className="Card"
    >
      <h3>
        First card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Second card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
    <div
      className="Card"
    >
      <h3>
        Third card
      </h3>
      <p>
        lorem ipsum
      </p>
    </div>
  </div>
</section>
`;
